src/DnsSecStack.ts

```diff
@@ -0,0 +1,98 @@
+import * as cdk from 'aws-cdk-lib';
+import { aws_ssm as SSM, Tags, aws_iam as IAM, aws_kms as KMS } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { Statics } from './Statics';
+
+export interface DnsSecStackProps extends cdk.StackProps {
+  /**
+   * DNSSEC KMS key alias
+   */
+  alias: string;
+}
+
+export class DnsSecStack extends cdk.Stack {
+
+  constructor(scope: Construct, id: string, props: DnsSecStackProps) {
+    super(scope, id, props);
+
+    Tags.of(this).add('cdkManaged', 'yes');
+    Tags.of(this).add('Project', Statics.projectName);
+
+    // Create the (expensive $) key
+    const dnssecKey = this.addDNSSecKey(props.alias);
+
+    // Store the key arn in a parameter for use in other projects
+    //   (note that this parmeter is in us-east-1)
+    new SSM.StringParameter(this, 'account-dnssec-kms-key-arn', {
+      stringValue: dnssecKey.keyArn,
+      parameterName: Statics.accountDnsSecKmsKey,
+    });
+
+  }
+
+  /**
+   * Constructs a key using KMS
+   * @returns the newy kms key
+   */
+  addDNSSecKey(alias: string) {
+    const dnssecKmsKey = new KMS.Key(this, 'dnssec-kms-key', {
+      keySpec: KMS.KeySpec.ECC_NIST_P256,
+      keyUsage: KMS.KeyUsage.SIGN_VERIFY,
+      policy: new IAM.PolicyDocument({
+        statements: [
+          new IAM.PolicyStatement({
+            actions: ['kms:Sign'],
+            principals: [new IAM.AccountRootPrincipal()],
```

Comment:
Checken of AccountRootPrincipal voldoende is

Reply:
Ik geloof dat dit voldoende is. De key mag benaderd worden door route53 overal in het account waar de key aangemaakt wordt. Dat is de enige service die er bij zou moeten kunnen.